firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistributionTesterApiClient.java

```
@@ -0,0 +1,136 @@
+  private static final String DISPLAY_VERSION_JSON_KEY = "displayVersion";
+  private static final String RELEASE_NOTES_JSON_KEY = "releaseNotes";
+  private static final String BINARY_TYPE_JSON_KEY = "binaryType";
+  public static final int DEFAULT_BUFFER_SIZE = 8192;
+
+  public AppDistributionRelease fetchLatestRelease(
+      @NonNull String fid, @NonNull String appId, @NonNull String apiKey, @NonNull String authToken)
+      throws FirebaseAppDistributionException, ProtocolException {
+
+    AppDistributionRelease latestRelease = null;
+    HttpsURLConnection conn = openHttpsUrlConnection(appId, fid);
+    conn.setRequestMethod(REQUEST_METHOD);
+    conn.setRequestProperty(API_KEY_HEADER, apiKey);
+    conn.setRequestProperty(INSTALLATION_AUTH_HEADER, authToken);
+
+    try {
+      Log.v(TAG, String.valueOf(conn.getResponseCode()));
+      JSONObject latestReleaseJson = readFetchReleaseInputStream(conn.getInputStream());
+      Log.v("Json release", latestReleaseJson.toString());
+
+      long latestBuildVersion = Long.parseLong(latestReleaseJson.getString(BUILD_VERSION_JSON_KEY));
+      final String displayVersion = latestReleaseJson.getString(DISPLAY_VERSION_JSON_KEY);
+      final String buildVersion = latestReleaseJson.getString(BUILD_VERSION_JSON_KEY);
+      String releaseNotes;
+      try {
+        releaseNotes = latestReleaseJson.getString(RELEASE_NOTES_JSON_KEY);
+      } catch (JSONException e) {
+        releaseNotes = "";
+      }
+      final BinaryType binaryType =
+          latestReleaseJson.getString(BINARY_TYPE_JSON_KEY).equals("APK")
+              ? BinaryType.APK
+              : BinaryType.AAB;
+
+      latestRelease =
+          AppDistributionRelease.builder()
+              .setDisplayVersion(displayVersion)
+              .setBuildVersion(buildVersion)
+              .setReleaseNotes(releaseNotes)
+              .setBinaryType(binaryType)
+              .build();
+
+    } catch (IOException | JSONException e) {
```

we should return a NETWORK FAILURE instead of "UNKNOWN" error